Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -1,17 +1,41 @@
 import Foundation
 import TSCBasic
 
-public final class GraphTraverser: GraphTraversing {
+public final class GraphTraverser {
     private let graph: Graph
     public var name: String { graph.name }
     public var hasPackages: Bool { !graph.packages.isEmpty }
     public var path: AbsolutePath { graph.entryPath }
     public var workspace: Workspace { graph.workspace }
-    public var projects: [AbsolutePath: Project]
+    public let projects: [AbsolutePath: Project]
+    public let targets: [AbsolutePath: [String: Target]]
 
     public init(graph: Graph) {
         self.graph = graph
         projects = Dictionary(uniqueKeysWithValues: graph.projects.map { ($0.path, $0) })
+        targets = graph.targets.mapValues { (targets: [TargetNode]) -> [String: Target] in
+            Dictionary(uniqueKeysWithValues: targets.map { ($0.name, $0.target) })
+        }
+    }
+
+    public func rootTargets() -> Set<ValueGraphTarget> {
+        Set(graph.entryNodes.compactMap { $0 as? TargetNode }
+            .map { ValueGraphTarget(path: $0.path, target: $0.target, project: $0.project) })
+    }
+
+    public func rootProject() -> Project? {
+        projects[graph.entryPath]
+    }
```

Comment:
it doesn't look like this is used anywhere - possibly left over from a previous iteration

Reply:
That's a good point. I'll delete it.